src/storage/v2/replication/replication_client.cpp

```diff
@@ -233,7 +239,12 @@ bool ReplicationStorageClient::FinalizeTransactionReplication(Storage *storage,
   // valid during a single transaction replication (if the assumption
   // that this and other transaction replication functions can only be
   // called from a one thread stands)
+  spdlog::trace("Finalizing transaction on replica {} in state {}", client_.name_, StateToString());
   if (State() != replication::ReplicaState::REPLICATING) {
+    // TODO(antoniofilipovic) This was missing as we could have been in RECOVERY STATE
+    // because we missed a transaction, and we need to finalize current transaction
+    spdlog::trace("Skipping finalizing transaction on replica {} because it's not replicating", client_.name_);
+    replica_stream_.reset();
```

Comment:
Let's say 1 transaction runs and the immediately after the other starts. Due to other starting we switch to the RECOVERY state. We then need to do RECOVERY, we do the recovery we are in a READY state. After starting transaction again, we haven't cleared replica_stream

Reply:
False assumption, removing this